### ISSUE TITLE
SAM ManagedPolicyArns Issue

### ISSUE THREAD

**Issue by vamegh** [aws-cloudformation/cfn-lint#2419]
### CloudFormation Lint Version

cfn-lint 0.67.0

### What operating system are you using?

Kubuntu

### Describe the bug

E3031 ManagedPolicyArns contains invalid characters (Pattern: arn:(aws[a-zA-Z-]*)?:iam::(\d{12}|aws):policy/[a-zA-Z_0-9+=,.@\-_/]+)

The policy in question is: VPCAccessPolicy

Which according to the docs: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-policy-template-list.html#vpc-access-policy is a Valid policy, however the linter does not pick up on this. 

### Expected behavior

There should be no issue here. The linter should accept the policy

### Reproduction template

```yaml
LambdaFunction:
    Type: AWS::Serverless::Function

    Properties:
      FunctionName: "Some Function"
      Handler: some.handler
      Runtime: java
      PackageType: Zip
      CodeUri: whatever
      Architectures:
        - arm64
      MemorySize: 2048
      Tracing: Active
      Policies:
        - VPCAccessPolicy
```

**Comment by vamegh** [comment]
Going by docs: These are all SAM managed policies available: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-policy-templates.html

Which comes from here: https://github.com/aws/serverless-application-model/blob/develop/samtranslator/policy_templates_data/policy_templates.json

similar issue reported in the past: https://github.com/aws-cloudformation/cfn-lint/issues/2022 however, the fix applied then does not fix the issue I have raised above.

**Comment by kddejong** [comment]
I don't see `VPCAccessPolicy` as a managed policy.  

Looks like the policy is defined in the SAM code.  https://github.com/aws/serverless-application-model/blob/579353c793b9c779c0f869f7bb7e4a16b16f8767/samtranslator/policy_templates_data/policy_templates.json#L672-L691

The resulting section from the translator is:
```
"ManagedPolicyArns": [
     "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
     "arn:aws:iam::aws:policy/AWSXrayWriteOnlyAccess",
     "VPCAccessPolicy"
    ],
```

**Comment by kddejong** [comment]
According to the [docs](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-policy-templates.html) this is a SAM IAM policy template.

This is the correct way to use SAM policy templates:
```
Transform: AWS::Serverless-2016-10-31
Resources:
  LambdaFunction:
    Type: AWS::Serverless::Function

    Properties:
      FunctionName: "Some Function"
      Handler: some.handler
      Runtime: java11
      PackageType: Zip
      CodeUri: whatever
      Architectures:
        - arm64
      MemorySize: 2048
      Tracing: Active
      Policies:
        - VPCAccessPolicy: {}
```

**Comment by kddejong** [comment]
There is a possible rule here that would make this issue more clear.
